site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+
+For equality deletes, all the equality predicates must be loaded to memory to check against every row in data files.
+Therefore, having too many equality deletes can have the risk of running out of memory.
+It is possible to use an external storage like RocksDB to store equality predicates, but too many equality deletes would still have huge impact to the read performance.
+
+Because of this difference, position deletes tend to be more efficient in reading than equality deletes.
+This has an impact to our delete compaction strategies.
+
+### MoR delete compaction
+
+After being able to read and write delete files, users need to run delete compactions to optimize the read performance.
+There are 3 common Iceberg delete compactions:
+
+#### Rewrite position deletes (REWRITE)
+
+As we discussed in the [reader](#data-with-delete-reader) section, having fewer position deletes per data file is computationally more efficient.
+This compaction optimizes the positions deletes layout, such as combining all position deletes in a single partition to a single file.
+
+#### Convert equality deletes (CONVERT)
+
+As we discussed in the [reader](#data-with-delete-reader) section, equality deletes has the potential to introduce out-of-memory issue and should be removed as soon as possible.
+This compaction converts equality deletes to position deletes by performing a table scan for the predicates in the equality deletes to figure out the affected file and row positions.
+
+#### Merge deletes (MERGE)
+
+This compaction merges deletes into data files and removes the deletes.
+This is likely the definition of delete compaction that most people think of first.
+
+Note that Iceberg also offers [data compaction](../maintenance/#compact-data-files) as a separated concept to optimize data file layout.
+As a part of reading data files, the Iceberg reader already merges deletes into the new data file it produces,
+so data compaction also "merges" deletes, but it does not remove any delete file because the delete file might still reference some data files not rewritten.
+However, MERGE guarantees a delete file can be removed safely by rewriting all the data files affected by it.
+
+It is arguable that if a table is continuously compacted using data compaction, it is possible that for a delete file,
+all the data files it references are already rewritten, and there is no data file of a lower sequence number anymore.
+Therefore, instead of MERGE, a user can also choose to only run data compaction with an additional procedure that periodically expires delete files that do not reference any data files.
+
+However, there is one more feature that data compaction does not provide but MERGE provides, 
+that is to rewrite data files while preserving the sequence number.
+In data compaction, data files are always rewritten with a new sequence number.
```

Comment:
Why does data compaction have to rewrite with a new sequence number? Why can't it also do what MERGE is doing with respect to sequence numbers.